importer-db-consistency/scripts/EpochSlotToBlkHash.js

```diff
@@ -0,0 +1,84 @@
+var fetch = require("node-fetch");
+var lodash = require("lodash");
+
+const serverRoute = 'https://explorer.iohkdev.io/api';
+
+const slotsPerEpoch = 21600;
+
+const randomNumberWithMax = function (topNumber) {
+  // TODO: Improve random method used
+  return Math.floor((Math.random() * topNumber));
+}
+
+function removeNullFromArray(actual) {
+  var newArray = [];
+  for (var i = 0; i < actual.length; i++) {
+    if (actual[i]) {
+      newArray.push(actual[i]);
+    }
+  }
+  return newArray;
+}
+
+const generateRandomEpochAndSlots = function (topEpoch, numGenerated) {
+  const epochAndSlotGenerated = [];
+  for (let i = 0; i < numGenerated; i++) {
+    const epochGen = randomNumberWithMax(Number(topEpoch) + 1);
+    const slotGen = randomNumberWithMax(slotsPerEpoch);
+    epochAndSlotGenerated.push([epochGen, slotGen]);
+  }
+  return epochAndSlotGenerated;
+};
+
+const parseResponse = function (response) {
+  return response.json();
+};
+
+const handleErrors = function (responseJson) {
+  if (responseJson.error) {
+    console.error(`[EpochSlotToBlkHash] error[${responseJson.error.name}]`);
+    throw responseJson.error;
+  } else {
+    return Promise.resolve(responseJson);
+  }
+};
+
+const epochAndSlotToBlkHash = async function (epochNumber, slotNumber) {
+  return fetch(`${serverRoute}/epochs/${epochNumber}/${slotNumber}`, {
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json;charset=utf-8',
+    },
+  })
+  .then(parseResponse)
+  .then(handleErrors)
+  .then(response => {
+    try {
+      const { Right: [{ cbeBlkHash }] } = response;
+      return cbeBlkHash;
+    } catch (e) {
+      console.error(`Epoch ${epochNumber} and slot ${slotNumber} have no block`);
+      return null;
+    }
+  }).catch((error) => {
+    console.error('Explorer request failed due to ', error);
+  });
+};
+
+//NOTE: includes topEpoch in the possible epochs to be generated
+const getRandomBlkHashes = function (topEpoch, numGenerated) {
+  const epochAndSlotGen = generateRandomEpochAndSlots(topEpoch, numGenerated);
+  
+  // FIXME: Do requests in batches to not kill the explorer
```

Comment:
fixme?

Reply:
nvm just noticed you mentioned you double-checked all of these already.

Reply:
Regarding this file and everything related with this random block external consistency test, do you think that it would be better to remove it instead?

The original idea for the consistency test for the tx history was checking random blocks, however, too many requests had to be done to the explorer (to translate from slot and epoch to block hash) which slowed it down (and killed the explorer in the process if no batching was used).

This was replaced with sequential checking of the tx history (using `./postgres-consistency/scripts/launch/external-consistency-from-blk.sh`) but I left the progress for checking random blocks in case it was needed later. WDYT?

Reply:
As long as it doesn't cause additional merging problems in the future I don't really mind keeping it. That being said, if it's not currently being used probably it will never be used after the hand-off (since realistically people tend not to go through unused piece of legacy codebases) so I'm not opposed to removing it either.

Reply:
Removed this consistency checked as it's not used!